homeassistant/components/switch/xiaomi_vacuum.py

```diff
@@ -22,7 +25,41 @@
     vol.Optional(CONF_NAME): cv.string,
 })
 
-REQUIREMENTS = ['python-mirobo==0.0.8']
+REQUIREMENTS = ['python-mirobo==0.1.0']
+
+ATTR_COMMAND = 'command'
+ATTR_PARAMS = 'params'
+SERVICE_COMMAND = 'xiaomi_vacuum_command'
+
+ATTR_FANSPEED = 'fanspeed'
+SERVICE_FANSPEED = 'xiaomi_vacuum_set_fanspeed'
+
+SERVICE_START_REMOTE_CONTROL = 'xiaomi_vacuum_start_remote_control'
+SERVICE_MOVE_REMOTE_CONTROL = 'xiaomi_vacuum_move_remote_control'
+SERVICE_STOP_REMOTE_CONTROL = 'xiaomi_vacuum_stop_remote_control'
+SERVICE_REMOTE_CONTROL = 'xiaomi_vacuum_remote_control'
+
+ATTR_RC_VELOCITY = 'velocity'
+ATTR_RC_ROTATION = 'rotation'
+ATTR_RC_DURATION = 'duration'
+
+COMMAND_SERVICE_SCHEMA = vol.Schema({
+    vol.Required(ATTR_COMMAND): cv.string,
+    vol.Optional(ATTR_PARAMS): cv.string,
+})
+
+FANSPEED_SERVICE_SCHEMA = vol.Schema({
+    vol.Required(ATTR_FANSPEED): vol.All(vol.Coerce(int),
+                                         vol.Any(38, 60, 77, 90)),
+})
+
+REMOTE_CONTROL_SERVICE_SCHEMA = vol.Schema({
+    vol.Optional(ATTR_RC_VELOCITY): vol.All(vol.Coerce(float),
+                                            vol.Range(min=-0.3, max=0.3)),
+    vol.Optional(ATTR_RC_ROTATION): vol.All(vol.Coerce(int),
+                                            vol.Range(min=-180, max=180)),
```

Comment:
min & max are 179, not 180. As those are not exact degrees, I'm sure where this should be enforced, though.